express/storage_mount.py

```diff
@@ -0,0 +1,102 @@
+"""Storage mounting functionalities"""
+import subprocess  # nosec
+import logging
+import os
+from typing import Sequence, List, NamedTuple, Dict, Optional
+from enum import Enum
+from pathlib import Path
+
+from express.io import create_subprocess_arguments
+
+logger = logging.getLogger(__name__)
+
+
+class CloudStorageConfig(NamedTuple):
+    """
+    Defines the configuration for a specific cloud provider's storage system.
+
+    Attributes:
+        storage_prefix (str): The prefix used to access the storage system.
+        mount_command (str): The command used to mount the storage system.
+    """
+
+    storage_prefix: str
+    mount_command: str
+
+
+class CloudProvider(Enum):
+    """
+    Enumerates the different cloud providers that can be used for remote storage.
+
+    Attributes:
+        GCP (CloudStorageConfig): Google Cloud Platform storage configuration.
+        AWS (CloudStorageConfig): Amazon Web Services storage configuration.
+        AZURE (CloudStorageConfig): Microsoft Azure storage configuration.
+    """
+
+    GCP = CloudStorageConfig("gs://", "gcsfuse")
+    AWS = CloudStorageConfig("s3://", "s3fs")
+    # TODO: double check azure storage prefix
+    AZURE = CloudStorageConfig("blob.core.windows.net://", "blobfuse")
+
+
+def mount_remote_storage(
+    mount_buckets: Sequence[str],
+    mount_dir: str,
+    mount_command: str,
+    mount_args: Optional[List[str]] = None,
+    mount_kwargs: Optional[Dict[str, any]] = None,
+):
+    """
+    Function that mounts a remote bucket to a local directory using FUSE
+    Args:
+        mount_buckets (str): the name of the bucket(s) to be mount
+        mount_dir (str): the local directory where to mount the buckets to
+        mount_command (str): the command used to mount the storage system
+        mount_args (Optional[List[str]]): Additional arguments to pass to the FUSE command.
```

Comment:
Which args and kwargs do we expect the user might want to pass here?

Reply:
There are quite a few extra arguments that can be passed to the fuse commands (e.g. passing in keys, enable additional logging, enable debugging). And I think they are all different depending on which cloud you use for fuse. We can maybe restrict the options once we figure out how we want to handle the authentication and enable other ones with extra argument flags that we pass in the function

Reply:
Question is which ones we want to expose to the user and which one we want to configure.